biogeophys/EDSurfaceAlbedoMod.F90

```diff
@@ -195,756 +125,918 @@ subroutine ED_Norman_Radiation (nsites, sites, bc_in, bc_out )
+      !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+      cosz = max(0.001_r8, currentPatch%solar_zenith_angle ) !copied from previous radiation code...
+      do ft = 1,numpft
+         sb = (90._r8 - (acos(cosz)*180/pi)) * (pi / 180._r8)
+         chil = xl(ft) !min(max(xl(ft), -0.4_r8), 0.6_r8 )
+         if ( abs(chil) <= 0.01_r8) then
+            chil  = 0.01_r8
+         end if
+         phi1b(ft) = 0.5_r8 - 0.633_r8*chil - 0.330_r8*chil*chil
+         phi2b(ft) = 0.877_r8 * (1._r8 - 2._r8*phi1b(ft)) !0 = horiz leaves, 1 - vert leaves.
+         gdir = phi1b(ft) + phi2b(ft) * sin(sb)
+         !how much direct light penetrates a singleunit of lai?
+         k_dir(ft) = clumping_index(ft) * gdir / sin(sb)
+      end do !FT
+         
+
+
+
+      !do this once for one unit of diffuse, and once for one unit of direct radiation
+      do radtype = 1, n_rad_stream_types 
+
+
+         ! Extract information that needs to be provided by ED into local array.
+         ! RGK: NOT SURE WHY WE NEED FTWEIGHT ...
+         ! ------------------------------------------------------------------------------
+
+         ftweight(:,:,:) = 0._r8
+         do L = 1,currentPatch%NCL_p
+            do ft = 1,numpft
+               do  iv = 1, currentPatch%nrad(L,ft)
+                  !this is already corrected for area in CLAP
+                  ftweight(L,ft,iv) = currentPatch%canopy_area_profile(L,ft,iv) 
+               end do  !iv
+            end do  !ft1
+         end do  !L
+         if (sum(ftweight(1,:,1))<0.999_r8)then
+            write(fates_log(),*) 'canopy not full',ftweight(1,:,1)
+         endif
+         if (sum(ftweight(1,:,1))>1.0001_r8)then
+            write(fates_log(),*) 'canopy too full',ftweight(1,:,1)
+         endif
+              
+         do L = 1,currentPatch%NCL_p !start at the top canopy layer (1 is the top layer.)
+
+            weighted_dir_tr(L)                 = 0.0_r8
+            weighted_fsun(L)                   = 0._r8
+            weighted_dif_ratio(L,1:hlm_numSWb) = 0._r8
+            
+            !Each canopy layer (canopy, understorey) has multiple 'parallel' pft's
+
+            do ft =1,numpft
+
+               if (currentPatch%canopy_mask(L,ft) == 1)then !only do calculation if there are the appropriate leaves.
+                  !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+                  ! Diffuse transmittance, tr_dif, do each layer with thickness elai_z.
+                  ! Estimated do nine sky angles in increments of 10 degrees
+                  ! PFT specific...
+                  !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+                  tr_dif_z(L,ft,:) = 0._r8
+                  do iv = 1,currentPatch%nrad(L,ft)
+                     do j = 1,9
+                        angle = (5._r8 + dble(j - 1) * 10._r8) * 3.142_r8 / 180._r8
```

Comment:
what's 3.142_r8?

Reply:
this is code is all copied and pasted, so I didn't do too much of a pass on numerics and constants and such.
 I essentially just pulled all the code in the norman radiation routine that was inside the patch loop, and made it a subroutine.  The only things that are different, are how the variables forc_dir and forc_dif are constructed, and the bc_  variables are processed before the patch level subroutine.

Reply:
yeah i know, just giving a hard time.  but if it is okay not to be b4b, no reason not to clean this stuff up while we're looking at it.